Upload to S3 Failing

- CTFd Version/Commit: 3.6.1
- Operating System: Linux (Docker container)
- Web Browser and Version: Chrome

**What happened?**

Upgrading CTFd resulting in S3 file uploads beginning to return 400 (bad request) status codes. I see one of the fixes for 3.6.1 was for S3, so perhaps a new bug was introduced.

Here are some additional facts which may be helpful:

  - The files are successfully making there way into S3, despite the error
  - The timezone I have configured for my server is CST

I can also confirm that my deployment had working file upload before upgrade to version 3.6.1 (file upload was working for 3.6.0).

**What did you expect to happen?**

File upload to continue working.

**How to reproduce your issue**

Deploy CTFd free version using version 3.6.1 with S3 file upload configured.

**Any associated stack traces or error logs**

The browser request returns error (400 status code):

```
{
    "success": false,
    "errors": {
        "location": [
            "I/O operation on closed file."
        ]
    }
}
```

The backend error is:

```
[ERROR] Error handling request
Traceback (most recent call last):
File "/opt/venv/lib/python3.9/site-packages/gunicorn/workers/base_async.py", line 113, in handle_request
resp.write_file(respiter)
File "/opt/venv/lib/python3.9/site-packages/gunicorn/http/wsgi.py", line 385, in write_file
if not self.sendfile(respiter):
File "/opt/venv/lib/python3.9/site-packages/gunicorn/http/wsgi.py", line 375, in sendfile
self.sock.sendfile(respiter.filelike, count=nbytes)
File "/opt/venv/lib/python3.9/site-packages/gevent/_socket3.py", line 486, in sendfile
return self._sendfile_use_send(file, offset, count)
File "/opt/venv/lib/python3.9/site-packages/gevent/_socket3.py", line 416, in _sendfile_use_send
self._check_sendfile_params(file, offset, count)
File "/opt/venv/lib/python3.9/site-packages/gevent/_socket3.py", line 461, in _check_sendfile_params
raise ValueError(
ValueError: count must be a positive integer (got 0)
```

--- a/CTFd/utils/uploads/__init__.py
+++ b/CTFd/utils/uploads/__init__.py
@@ -44,10 +44,11 @@ def upload_file(*args, **kwargs):
         model = PageFiles
         model_args["page_id"] = page_id
 
+    # Hash is calculated before upload since S3 file upload closes file object
+    sha1sum = hash_file(fp=file_obj)
+
     uploader = get_uploader()
     location = uploader.upload(file_obj=file_obj, filename=filename, path=parent)
-
-    sha1sum = hash_file(fp=file_obj)
 
     model_args["location"] = location
     model_args["sha1sum"] = sha1sum

